src/parsers.rs

```diff
@@ -1,13 +1,276 @@
 //! This module implements Temporal Date/Time parsing functionality.
 
+use crate::{
+    options::{DisplayCalendar, DisplayOffset, DisplayTimeZone},
+    Sign, TemporalError, TemporalResult, TemporalUnwrap,
+};
 use alloc::format;
-
-use crate::{TemporalError, TemporalResult, TemporalUnwrap};
-
 use ixdtf::parsers::{
     records::{Annotation, DateRecord, IxdtfParseRecord, TimeRecord, UtcOffsetRecordOrZ},
     IxdtfParser,
 };
+use writeable::{impl_display_with_writeable, Writeable};
+
+// TODO: Move `Writeable` functionality to `ixdtf` crate
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Precision {
+    Auto,
+    Minute,
+    Digit(u8),
+}
+
+pub struct FormattableTime {
+    pub hour: u8,
+    pub minute: u8,
+    pub second: u8,
+    pub nanosecond: u32,
+    pub precision: Precision,
+    pub include_sep: bool,
+}
+
+impl Writeable for FormattableTime {
+    fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
+        write_padded_u8(self.hour, sink)?;
+        if self.include_sep {
+            sink.write_char(':')?;
+        }
+        write_padded_u8(self.minute, sink)?;
+        if self.precision == Precision::Minute {
+            return Ok(());
+        }
+        if self.include_sep {
+            sink.write_char(':')?;
+        }
+        write_padded_u8(self.second, sink)?;
+        if self.nanosecond == 0 {
+            return Ok(());
+        }
+        sink.write_char('.')?;
+        write_nanosecond(self.nanosecond, self.precision, sink)?;
+
+        Ok(())
+    }
+}
+
+pub struct FormattableUtcOffset {
+    pub show: DisplayOffset,
+    pub offset: UtcOffset,
+}
+
+pub enum UtcOffset {
+    Z,
+    Offset(FormattableOffset),
+}
+
+impl Writeable for FormattableUtcOffset {
+    fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
+        if self.show == DisplayOffset::Never {
+            return Ok(());
+        }
+        match &self.offset {
+            UtcOffset::Z => sink.write_char('Z'),
+            UtcOffset::Offset(offset) => offset.write_to(sink),
+        }
+    }
+}
+
+fn write_padded_u8<W: core::fmt::Write + ?Sized>(num: u8, sink: &mut W) -> core::fmt::Result {
+    // NOTE:
+    if num < 10 {
+        sink.write_char('0')?;
+    }
+    num.write_to(sink)
+}
+
+fn write_nanosecond<W: core::fmt::Write + ?Sized>(
+    nanoseconds: u32,
+    precision: Precision,
+    sink: &mut W,
+) -> core::fmt::Result {
+    if nanoseconds > 1_000_000_000 {
+        return Err(core::fmt::Error);
+    }
```

Comment:
Nit: You should avoid returning `core::fmt::Error` because it gets turned into an unhandled `panic!` in most call sites. You should try and ensure that your nanoseconds are in range by adding invariants to the type being formatted, instead of returning an error in the Writeable impl.

If you need to return errors from the Writeable impl, consider using TryWriteable instead.